R/knitr-hooks.R

```diff
@@ -191,8 +243,24 @@ install_knitr_hooks <- function() {
           preserved_options$exercise.df_print <- "default"
         preserved_options$exercise.timelimit <- options$exercise.timelimit
         preserved_options$exercise.setup <- options$exercise.setup
-        preserved_options$exercise.checker <- deparse(options$exercise.checker)
 
+        root_chunk <- get_knitr_chunk(options$label)
+        root_options <- attr(root_chunk, "chunk_opts")
+        preserved_options$engine <- root_options$engine
+
+        # we need to rev so that we have correct order of setup chunks
+        setup_chunks <- rev(find_parent_setup_chunks(options))
```

Comment:
The comment above has the parent first, no need to `rev()` here
```suggestion
        setup_chunks <- find_parent_setup_chunks(options)
```